libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp

```diff
@@ -245,7 +384,48 @@ void AP_CRSF_Telem::process_ping_frame(ParameterPingFrame* ping)
     }
 
     _param_request.origin = ping->origin;
-    _request_pending = AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAM_DEVICE_PING;
+    _pending_request.frame_type = AP_RCProtocol_CRSF::CRSF_FRAMETYPE_PARAM_DEVICE_PING;
+}
+
+// request for device info
+void AP_CRSF_Telem::process_device_info_frame(ParameterDeviceInfoFrame* info)
+{
+    debug("process_device_info_frame: %d -> %d", info->origin, info->destination);
+    if (info->destination != 0 && info->destination != AP_RCProtocol_CRSF::CRSF_ADDRESS_FLIGHT_CONTROLLER) {
+        return; // request was not for us
+    }
+    
+    // we are only interested in RC device info for firmware version detection
+    if (info->origin != 0 && info->origin != AP_RCProtocol_CRSF::CRSF_ADDRESS_CRSF_RECEIVER) {
+        return; 
+    }
+    /*
+        Payload size is 58:
+        char[] Device name ( Null-terminated string, max len is 42 )
+        uint32_t Serial number
+        uint32_t Hardware ID
+        uint32_t Firmware ID (0x00:0x00:0xAA:0xBB AA=major, BB=minor)
+        uint8_t Parameters count
+        uint8_t Parameter version number        
+    */
+    uint8_t offset = 0;
+    // get the terminator of the device name string
+    while (info->payload[offset] != 0x0 && offset++ < 42U);
+    /*
+        fw major ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 3rd byte of sw id = 11bytes
+        fw minor ver = offset + terminator (8bits) + serial (32bits) + hw id (32bits) + 4th byte of sw id = 12bytes
+    */
+    _crsf_version.major = info->payload[offset+11];
+    _crsf_version.minor = info->payload[offset+12];
+
+    // should we use rf_mode reported by link statistics?
+    if (_crsf_version.major >= 3 && _crsf_version.minor >= 72) {
+        _crsf_version.use_rf_mode = true;
+    }
+
+    gcs().send_text(MAV_SEVERITY_DEBUG,"CRSF: fw version %d.%02d", _crsf_version.major, _crsf_version.minor);
```

Comment:
Do we want this?

Reply:
you mean do we really need to have crsf version info?
Well no, but it's cool to have it

Reply:
thinking I can save a status text and send the version when I notify custom telemetry init done